react/details/src/components/ConferenceDetailsContainer.js

```diff
@@ -15,42 +17,83 @@ class ConferenceDetailsContainer extends React.Component {
     this.state = {
       loading: true,
       conference: {},
-      error: null,
+      notificationStatus: Notification.INFO,
+      notificationMessage: null,
     };
 
     this.theme = createMuiTheme();
     this.closeNotification = this.closeNotification.bind(this);
+    this.fetchData = this.fetchData.bind(this);
   }
 
   componentDidMount() {
-    const { t, id, onError } = this.props;
+    const { authInitialized, authenticated } = this.props;
+    const userAuthenticated = authInitialized && authenticated;
+    if (userAuthenticated) {
+      this.fetchData();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { authInitialized, authenticated } = this.props;
+
+    const userAuthenticated = authInitialized && authenticated;
+    const changedAuth = prevProps.authInitialized !== authInitialized;
+
+    if (userAuthenticated && changedAuth) {
+      this.fetchData();
+    }
+  }
+
+  fetchData() {
+    const { id, onError, t, authenticated, authToken } = this.props;
 
-    if (id) {
-      getConference({ id })
-        .then(response => this.setState({ error: null, conference: response }))
+    if (id && authenticated) {
+      getConference({ id }, authToken)
+        .then(conference =>
+          this.setState({
+            notificationStatus: Notification.INFO,
+            notificationMessage: null,
+            conference,
+          })
+        )
         .catch(e => {
           onError(e);
-          this.setState({ error: t('common.couldNotFetchData') });
+          this.setState({
+            notificationStatus: Notification.ERROR,
+            notificationMessage: t('common.couldNotFetchData'),
+          });
         })
         .finally(() => this.setState({ loading: false }));
     } else {
-      this.setState({ loading: false, error: t('common.missingId') });
+      this.setState({
+        loading: false,
+        notificationStatus: Notification.ERROR,
+        notificationMessage: t('common.missingId'),
+      });
     }
   }
 
   closeNotification() {
-    this.setState({ error: null });
+    this.setState({
+      notificationStatus: Notification.INFO,
+      notificationMessage: null,
+    });
   }
 
   render() {
-    const { conference, error, loading } = this.state;
-    const { t } = this.props;
+    const { conference, notificationStatus, notificationMessage, loading } = this.state;
+    const { authenticated, t } = this.props;
 
     return (
       <ThemeProvider theme={this.theme}>
```

Comment:
what about moving `ThemeProvider`in the web component, like we did with the form widget?

Reply:
Would it be possible to be done with a separate task? I would like to focus on keycloak with this one